### DISCUSSION TITLE
connection timeout setting?

### DISCUSSION THREAD

**Discussion by bradbeshaw** [tdslite/tdslite#39] [Q&A] [answered]
Is there a connection timeout setting for [auto cr = driver.connect(params);]  ?

**Comment by xmkg** [comment, marked as answer]
> Is there a connection timeout setting for [auto cr = driver.connect(params);]  ?
> 

Hi @bradbeshaw 

There is no timeout setting at the moment, but the code has a implicit timeout mechanism that currently set to ~30 seconds. Feel free to create feature request if having a timeout setting would be useful for you.

https://github.com/tdslite/tdslite/blob/main/src%2Ftdslite-net%2Farduino%2Ftdsl_netimpl_arduino.hpp#L125-L138

**Comment by bradbeshaw** [comment]
Hello to you Mustafa,

I would like to make a feature request to include a timeout setting when attempting to connect.
Perhaps it could be an optional parameter passed via the dirver.connect command.
In my application for example, the ability to connect to the database is an environmental attribute that the program must be responsive to by facilitating a local storage option while the sensors continue to function. When connection to the remote data becomes viable, the program would then transmit it's local cache to that system.
Currently, if the remote data connection is available the tdslite library connects immediately.
If the remote data is not available the program remains unresponsive during this timeout/retry period.

Please do let me know if there are any questions regarding this request or if I may be of any assistance .

Regards

**Reply by xmkg** [reply]
Hello @bradbeshaw 

This should be fixed with the 0.5.0 release that I just published. It may take some time for Arduino library index to pick up the new release. Meanwhile, you can download it manually from the releases page: https://github.com/tdslite/tdslite/releases/tag/v0.5.0

Usage:

https://github.com/tdslite/tdslite/blob/317703b19eb28aef51dcc06593fa7acfe1295918/examples/arduino/03-select-rows/03-select-rows.ino#L189-L199
